cmd/memr/output.go

```diff
@@ -24,13 +24,23 @@ func S3Writer(reader io.ReadCloser, compress bool, region, bucket, key string, c
 		return nil, err
 	}
 
+	// Set size of multi part upload; by default the minimal is 5Mb
+	// A lower value here will have a lesser impact on memory pressure, and should be considered
+	// Increasing this value will directly impact how much memory we use
+	// Minimal size when possible will allow max 50GB memory size (5MB * 10.000)
+	var partSize int64
+	if uint64(manager.MaxUploadParts) * uint64(manager.MinUploadPartSize) > memory_size {
+		partSize = manager.MinUploadPartSize
+	} else {
+		// For bigger memory than 50GB, we calculate the size of the part
+		// part size = (memory size / max upload parts) + 1MB
+		partSize = int64(memory_size / uint64(manager.MaxUploadParts)) + (1024 * 1024)
+	}
```

Comment:
```suggestion
	const padSize = 1024 * 1024  // Use an extra mb as padding, just in case
	var partSize int64 = manager.MinUploadPartSize  // Default to minimum part size (5MB)
	// Check if: memory_size > (50GB - 1MB pad)
	if memory_size > (uint64(manager.MaxUploadParts) * uint64(manager.MinUploadPartSize)) + padSize {
		// For bigger memory than 50GB, we calculate the size of the part
		// part size = (memory size / max upload parts) + 1MB
		partSize = int64(memory_size / uint64(manager.MaxUploadParts)) + padSize
	}

Reply:
For this part, I believe the padSize should be added to the memory size, right?

```
memory_size + padSize > uint64(manager.MaxUploadParts) * uint64(manager.MinUploadPartSize)
```